processing/src/main/java/org/apache/carbondata/processing/store/writer/v3/CarbonFactDataWriterImplV3.java

```diff
@@ -106,23 +106,13 @@ public CarbonFactDataWriterImplV3(CarbonDataWriterVo dataWriterVo) {
       throws CarbonDataWriterException {
     // condition for writting all the pages
     if (!encodedTablePage.isLastPage()) {
-      boolean isAdded = false;
       // check if size more than blocklet size then write the page to file
       if (dataWriterHolder.getSize() + encodedTablePage.getEncodedSize() >= blockletSize) {
-        // if one page size is more than blocklet size
-        if (dataWriterHolder.getEncodedTablePages().size() == 0) {
-          isAdded = true;
-          dataWriterHolder.addPage(encodedTablePage);
-        }
-
         LOGGER.info("Number of Pages for blocklet is: " + dataWriterHolder.getNumberOfPagesAdded()
             + " :Rows Added: " + dataWriterHolder.getTotalRows());
         // write the data
         writeBlockletToFile();
       }
-      if (!isAdded) {
-        dataWriterHolder.addPage(encodedTablePage);
```

Comment:
For both cases (isAdded true or false), it will invoke `dataWriterHolder.addPage(encodedTablePage);`, so removing it.